vllm/entrypoints/openai/rpc/client.py

```diff
@@ -21,12 +25,40 @@
 # Time to wait before checking it the server process is alive.
 SERVER_START_TIMEOUT_MS = 1000
 
+# Inprocess path
+INPROC_PATH = f"inproc://{uuid4()}"
+
 
 class AsyncEngineRPCClient:
 
     def __init__(self, rpc_path: str):
         self.context = zmq.asyncio.Context()
-        self.rpc_path = rpc_path
+        self.context.set(zmq.constants.MAX_SOCKETS, VLLM_RPC_ZMQ_MAX_SOCKETS)
+
+        # PROXY
+        self.from_client = self.context.socket(zmq.constants.ROUTER)
+        self.from_client.bind(INPROC_PATH)
+
+        # Connection to RPC Server.
+        self.to_server = self.context.socket(zmq.constants.DEALER)
+        self.to_server.connect(rpc_path)
+
+        self.proxy_task = asyncio.create_task(
+            self.run_proxy(self.from_client, self.to_server))
+
+    async def run_proxy(self, socket_from, socket_to):
+        poller = zmq.asyncio.Poller()
+        poller.register(socket_from, zmq.constants.POLLIN)
+        poller.register(socket_to, zmq.constants.POLLIN)
+        while True:
+            events = await poller.poll()
+            events = dict(events)
+            if socket_from in events:
+                msg = await socket_from.recv_multipart()
+                await socket_to.send_multipart(msg)
+            elif socket_to in events:
```

Comment:
In theory it could have both
```suggestion
            if socket_to in events:
```